machine/util/callbacks/early_stopping_callback.py

```diff
@@ -0,0 +1,94 @@
+from machine.util.callbacks import Callback
+
+
+class EarlyStoppingCallback(Callback):
+    """
+    Original callback taken from https://github.com/ncullen93/torchsample
+    Early Stopping to terminate training early under certain conditions
+    """
+
+    def __init__(self,
+                 monitor='eval_losses',
+                 lm_name=None,
+                 min_delta=0,
+                 patience=5,
+                 minimize=True):
+        """
+        EarlyStopping callback to exit the training loop if training or
+        validation loss or metric does not improve by a certain amount
+        for a certain number of epochs
+        Args:
+            monitor : string in {'eval_losses', 'eval_metrics', 'train_losses', 'train_metrics'}
+                whether to monitor train or val loss
+            lm_name: loss or metric name eg. 'Avg NLLoss' or 'Word Accuracy'
```

Comment:
I find this name a bit unfortunate, because 'lm' is also commonly used to refer to language model. Luckily the docstrings are great, so I could easily understand what it is used for :). I'm not sure what would be a better name. Maybe just "objective"?

Reply:
Haha I also don't like the `lm` naming but I wanted something to describe both loss and metric - objective sounds much better - I'll change that!